### PR TITLE
make incompatibility of non-/usr-merged distros with --volatile=yes more discoverable

### PR DESCRIPTION
**Pull request by poettering**
/cc @teg @larskarlitski

### CODE REVIEW

File: src/nspawn/nspawn-mount.c [hunk 1]
```diff
@@ -1007,14 +1007,33 @@ static int setup_volatile_yes(
...
+
+                /* ENOENT is fine, just means the image is probably just a naked /usr and we can create the
+                 * rest. */
+        } else if (S_ISDIR(st.st_mode))
+                return log_error_errno(SYNTHETIC_ERRNO(EISDIR),
+                                       "Sorry, --volatile=yes mode is not supported with OS images that have not merged /bin/, /sbin/, /lib/, /lib64/ into /usr/. "
+                                       "Please work with your distribution and help them adopt the merged /usr scheme.");
+        else if (!S_ISLNK(st.st_mode))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Hmm, /bin in the OS image exists but is not a symlink. Something's wrong with the image.");
```
Comment (larskarlitski) [inline src/nspawn/nspawn-mount.c]: People probably won't see this error often, but when they do, it is not very helpful. I think something like "Error starting image: `/bin` must be a symlink (for tress with merged `/usr`) or a directory." would help me find the error quicker.

**Comment by poettering** [comment]
I force pushed a new version with an updated error msg. Didn't use @larskarlitski wording exactly though (couldn't figure out what "tress" was supposed to be). ptal!

File: src/nspawn/nspawn-mount.c [hunk 2]
```diff
@@ -1007,14 +1007,33 @@ static int setup_volatile_yes(
...
+
+                /* ENOENT is fine, just means the image is probably just a naked /usr and we can create the
+                 * rest. */
+        } else if (S_ISDIR(st.st_mode))
+                return log_error_errno(SYNTHETIC_ERRNO(EISDIR),
+                                       "Sorry, --volatile=yes mode is not supported with OS images that have not merged /bin/, /sbin/, /lib/, /lib64/ into /usr/. "
+                                       "Please work with your distribution and help them adopt the merged /usr scheme.");
+        else if (!S_ISLNK(st.st_mode))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Error starting image: if --volatile=yes is used /bin must be a symlink (for merged /usr support) or non-existant (in which case a symlink is created automatically).");
```
Comment (keszybz) [inline src/nspawn/nspawn-mount.c]: typo: non-existant

Comment (poettering) [reply src/nspawn/nspawn-mount.c]: huh? that's what it says?

Comment (keszybz) [reply src/nspawn/nspawn-mount.c]: Should be "non-existent"

Comment (poettering) [reply src/nspawn/nspawn-mount.c]: oh, uh, one day i'll get this into my head

**Comment by poettering** [comment]
force pushed new version now without any existant and existent typos (hopefully).

**Comment by poettering** [comment]
CI failure appears unrelated. Let's megre.

**Pull request merged**